lighthouse-core/gather/driver.js

```diff
@@ -643,18 +644,18 @@ class Driver {
             } else {
               log.verbose('Driver', `CPU has been idle for ${timeSinceLongTask} ms`);
               const timeToWait = waitForCPUQuiet - timeSinceLongTask;
-              lastTimeout = setTimeout(() => checkForQuiet(driver, resolve), timeToWait);
+              lastTimeout = setTimeout(() => checkForQuiet(driver, resolve, reject), timeToWait);
             }
           }
-        });
+        }).catch(e => reject(e));
     }
 
     /** @type {(() => void)} */
     let cancel = () => {
       throw new Error('_waitForCPUIdle.cancel() called before it was defined');
     };
     const promise = new Promise((resolve, reject) => {
-      checkForQuiet(this, resolve);
+      checkForQuiet(this, resolve, reject);
```

Comment:
what happens if reject is called twice? is that possible here?

Reply:
IIRC, once a promise has been resolved/rejected all further calls will be noops